lib/slack/web/faraday/response/raise_error.rb

```diff
@@ -3,28 +3,33 @@ module Slack
   module Web
     module Faraday
       module Response
-        class RaiseError < ::Faraday::Response::Middleware
+        class RaiseError < ::Faraday::Middleware
           def on_complete(env)
             raise Slack::Web::Api::Errors::TooManyRequestsError, env.response if env.status == 429
 
             return unless env.success?
+            return unless env.body
 
-            body = env.body
-            return unless body
+            body = process_body(env)
             return if body['ok']
 
-            error_message =
-              body['error'] || body['errors'].map { |message| message['error'] }.join(',')
-
+            error_message = body['error'] || body['errors'].map { |message| message['error'] }.join(',')
             error_class = Slack::Web::Api::Errors::ERROR_CLASSES[error_message]
             error_class ||= Slack::Web::Api::Errors::SlackError
             raise error_class.new(error_message, env.response)
+          # TODO: Do we need to still handle ::Faraday::ParsingError or will this be handled else where?
+          # rescue ::Faraday::ParsingError
```

Comment:
This parse rescue has been moved to `lib/slack/web/faraday/request.rb` as the `Faraday::ParsingError` is now triggered when processing the response.

As part of this, and to handle non JSON responses such as HTML 500s, I've currently set `content_type: /\b*$/` in the `:json` response options. The question is do you think this should be more restrictive and only parse certain known content types?

Reply:
This means content-type can be anything, doesn't it? So what's the point in having it? I think we should just try to parse any response because the Slack API says it will always send JSON, except when it's broken, and that's when we need this.